spec/src/main/asciidoc/rationale.asciidoc

```diff
@@ -0,0 +1,80 @@
+By infusing asynchronous messaging passing at the core of the system, applications enforcing the reactive system's characteristics are inherently elastic and resilient.
+
+image::reactive-system.png[Reactive System Characteristics, 400, 400]
+
+Microservices part of a reactive systems interacts using _messages_.
+The location and temporal decoupling, promoted by this interaction mechanism, exhibits numerous benefits such as:
+
+* Better failure management as the temporal decoupling would avoid message-loss and avoid cascading failures.
+* Improved elasticity as under fluctuating load the system can decide to scale up and down some of the microservices.
+* The ability to introduce new feature quickly as every component receives and publishes messages.
+
+The MicroProfile Reactive Messaging specification aims to deliver applications embracing the characteristics of reactive systems.
+
+=== On JMS and Message Driven Beans
+
+Java EE offers JMS and Message Driven Beans for handling asynchronous communication; however, there are some problems with these specifications:
+
+* Both are designed for a technology landscape where messaging was a heavyweight process used on the edge of the system to hand control of a transaction from system to another; consequently, there is a lot of boilerplate associated with using these technologies.
+* It is assumed their design that consistency is handled using distributed transactions. 
+However, modern message brokers, popular in microservice deployments, such as Apache Kafka, Amazon Kinesis and Azure Event Hubs, do not support XA transactions, rather, message acknowledgment is handled using offsets with at least once delivery guarantees.
+* They do not have support for asynchronous IO; it is assumed that message processing is done on a single thread, however, many modern specifications are moving to asynchronous IO.
+
+Hence a lighter weight, reactive solution to messaging is desirable for MicroProfile to ensure microservices written using MicroProfile are able to meet the demands required by the architecture.
+
+=== Use cases
+
+MicroProfile Reactive Messaging aims to provide a way to implement _event-driven_ microservices using MicroProfile. 
+The key characteristics of the specification make it versatile and perfectly suited to build different types of architecture and applications.
+
+First, asynchronous interactions with different services and resources can be implemented using Reactive Messaging. 
+Typically, asynchronous database drivers can be used in conjunction with Reactive Messaging to read and write into a data store in a non-blocking and asynchronous manner.
+
+When building microservices, the CQRS and event-sourcing patterns provide an answer to the data sharing between microservices. 
+Reactive Messaging can also be used as the foundation to CQRS and Event-Sourcing mechanism, as these patterns embrace message-passing as core communication pattern.
```

Comment:
I would like to include, if we can work out a good approach, the concepts of [Change Data Capture](https://en.wikipedia.org/wiki/Change_data_capture) as it might apply to @Outgoing to a datastore - if this data capture could be done at a level 'above' the low level database 'hooks' currently being used. 
Additionaly the concept of a [channel adapter](https://www.enterpriseintegrationpatterns.com/patterns/messaging/ChannelAdapter.html) roots this technology firmly in an established use case pattern.

Reply:
This is probably not wise to include at the moment for this version of the spec.